pyiode/pyiode_exec.pyx

```diff
@@ -6,18 +6,109 @@
 #  -----------------------
 # 
-#  TODO
-#  ---- 
-#   idt_execute(idt_list: Union(str, List[str]), sample: Union(str, List[str]), var_files: Union(str, List[str]), scl_files: Union(str, List[str])) | Execute a list of identities on a given sample
+#   idt_execute(sample: Optional[Union[str, List[str]]] = None, idt_list: Optional[Union[str, List[str]]] = None, var_files: Optional[Union[str, List[str]]] = None, scl_files: Optional[Union[str, List[str]]] = None,trace: int = 0) | Execute a list of identities on a given sample
 
+cdef extern from "iode.h": 
+    cdef int    IodeExecuteIdts(char *smpl, char *idt_list, char *var_files, char *scl_files, int trace)
 
-def idt_execute(idt_list: Union[str, List[str]], 
-                sample: Union[str, List[str]], 
-                var_files: Union[str, List[str]], 
-                scl_files: Union[str, List[str]]):
-    '''
-    Estimate a (block of) equations
-     
+def idt_execute(sample: Optional[Union[str, List[str]]] = None, 
+                idt_list: Optional[Union[str, List[str]]] = None, 
+                var_files: Optional[Union[str, List[str]]] = None, 
+                scl_files: Optional[Union[str, List[str]]] = None,
+                trace: bool = False):
+    r'''
+    Execute a list of identities
+    
+    Parameters
+    ----------
+    sample: Optional[Union[str, List[str]]] = None
+            range of period on which the identities must be calculated
+            if sample is None or empty, the WS sample is used 
+            
+    idt_list: Optional[Union[str, List[str]]] = None
+            list of identities to execute
+            if empty, all identities are executed
+            
+    var_files: Optional[Union[str, List[str]]] = None
+            list of files the needed variables must be read from. "WS" means current current Variable WS.
+            if empty, only the current KV_WS is used
+            
+    scl_files: Optional[Union[str, List[str]]] = None
+            list of files the needed scalars must be read from. "WS" means current current WS.
+            if empty, only the current workspace is used
+                        
+    trace: int = 0
+            optional trace indicating the source of the variables and scalars. For example:
+            
+            Execution of identities
+            
+            Parameters
+                Execution sample : 1960Y1:2015Y1
+            
+                Variables loaded
+                    From WS : AOUC AOUC_ COTRES DEBT DTF DTH EFMY EFXY EX FLG FLGR GAP2 GAP_ GOSF HF  
+                    
+                Scalars loaded
+                    From ../data/fun : gamma gamma2 gamma3 gamma4 knf3
+            
+    Examples
+    --------
+        # define the "trace" file
+        iode.w_dest("test_iode.htm", iode.W_HTML)
+    
+        # Simple case: no vars in WS, sample given
+        iode.ws_clear_var()
+        iode.ws_sample_set("1960Y1", "2015Y1")
+        iode.ws_load_idt(f"{IODE_DATA_DIR}fun")
```

Comment:
`IODE_DATA_DIR` is not yet defined in the iode module.
Please see pyiode_sample.pyx for example of how to import data

Here you should have 
```python
ws_load_idt("../data/fun.idt")
```

Reply:
TODO

Reply:
Added TODO in source code